File: src/main/java/dev/sky_lock/pocketlifevehicle/packet/ServerPacket.java
package dev.sky_lock.pocketlifevehicle.packet;

import com.comphenix.protocol.PacketType;
import com.comphenix.protocol.ProtocolLibrary;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

import java.lang.reflect.InvocationTargetException;
import java.util.logging.Level;

/**
 * @author sky_lock
 */

public class ServerPacket extends AbstractPacket {

    ServerPacket(PacketType type) {
        super(ProtocolLibrary.getProtocolManager().createPacket(type));
    }

    public void send(Player player) {
        try {
            ProtocolLibrary.getProtocolManager().sendServerPacket(player, getPacket());
        } catch (InvocationTargetException ex) {
            Bukkit.getLogger().log(Level.WARNING, "Could not send a packet");
        }
    }

    public void broadCast() {
        ProtocolLibrary.getProtocolManager().broadcastServerPacket(getPacket());
    }
}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/packet/ServerPacket.kt
package dev.sky_lock.pocketlifevehicle.packet

import com.github.retrooper.packetevents.PacketEvents
import com.github.retrooper.packetevents.wrapper.PacketWrapper
import org.bukkit.Bukkit
import org.bukkit.entity.Player

/**
 * @author sky_lock
 */

abstract class ServerPacket<T : PacketWrapper<T>>(private val wrapper: PacketWrapper<T>) {
    private val playerManager = PacketEvents.getAPI().playerManager

    fun send(player: Player) {
        playerManager.sendPacket(player, wrapper)
    }

    fun broadcast() {
        Bukkit.getOnlinePlayers().forEach { player -> send(player) }
    }
}